src/include/miopen/conv/problem_description.hpp

```diff
@@ -223,8 +253,20 @@ struct ProblemDescription
     // Weights getters
     miopenDataType_t GetWeightsDataType() const { return weights.GetType(); }
     std::size_t GetWeightsDepth() const { return GetD5(GetSpatialDims(), weights.GetLengths()); }
-    std::size_t GetWeightsHeight() const { return GetH5(GetSpatialDims(), weights.GetLengths()); }
-    std::size_t GetWeightsWidth() const { return GetW5(GetSpatialDims(), weights.GetLengths()); }
+    std::size_t GetWeightsHeight() const
+    {
+        if(weights.GetLayout_str() == "CHWNc")
+            return GetHofCHWN(weights.GetLengths());
+        else
+            return GetH5(GetSpatialDims(), weights.GetLengths());
```

Comment:
@jerryyin Would this work correctly with NCHW layout? I am afraid that some change here is necessary but missed in #343.

Reply:
This seems to be correct? Back in #343 We only supports NCHW or NCDHW alternatives. This additional branch to handle `CHWNc` seems to be what it should do in this PR.

Reply:
@jerryyin You're right. NHWC support was introduced in in #728. So does #728 missing changes in src/include/miopen/conv/problem_description.hpp, what do you think?

/cc @carlushuang

Reply:
#728 is far before this PR at that time there is no `GetLayout_str()` API. I'm not sure what is missing you refer to?

Reply:
@carlushuang My original question is about `GetWeightsHeight()` and similar functions. These were introduced 3 yrs ago for NCHW, and I would expect that some modification is necessary to support NHWC properly (in #728).

AFAICS now, the library requires (and supports) only the following order of elements in the lengths vector: `{ N,C,H,W }` (or `{ N,C,D,H,W }`), **_regardless of the actual layout_**. If so, then `GetWeightsHeight()` and similar functions do indeed support NHWC (no modifications to these functions were required in #728 to support NHWC).

Strictly speaking, this is off-topic in this PR. I can move the discussion there, if you wish.